Make run is broken

Make run fails with the following error:

```
INFO[0000] Versions                                      arch=amd64 jaeger-operator=v1.11.1-5-ged370cf operator-sdk=v0.5.0 os=linux version=go1.11.1
INFO[0000] Auto-detected the platform                    platform=kubernetes
INFO[0000] Automatically adjusted the 'es-provision' flag  es-provision=false
FATA[0000] unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined 
exit status 1
make: *** [Makefile:106: run] Error 
```

Running git bisect indicates that this commit https://github.com/jaegertracing/jaeger-operator/commit/ed370cf4fa5559d284f885420586293cd0229c80 introduced the problem cc) @objectiser.

--- a/pkg/cmd/start/main.go
+++ b/pkg/cmd/start/main.go
@@ -182,11 +182,14 @@ func start(cmd *cobra.Command, args []string) {
 	}
 
 	// Create Service object to expose the metrics port.
-	var operatorService *corev1.Service
-	if operatorService, err = metrics.ExposeMetricsPort(ctx, viper.GetInt32("metrics-port")); err != nil {
+	operatorService, err := metrics.ExposeMetricsPort(ctx, viper.GetInt32("metrics-port"))
+	if err != nil {
 		log.Fatal(err)
-	} else if err = setOwnerReference(operatorService); err != nil {
-		log.Fatal(err)
+	} else if operatorService != nil {
+		err = setOwnerReference(operatorService)
+		if err != nil {
+			log.Fatal(err)
+		}
 	}
 
 	// Start the Cmd

--- a/pkg/cmd/start/main.go
+++ b/pkg/cmd/start/main.go
@@ -186,8 +186,7 @@ func start(cmd *cobra.Command, args []string) {
 	if err != nil {
 		log.Fatal(err)
 	} else if operatorService != nil {
-		err = setOwnerReference(operatorService)
-		if err != nil {
+		if err := setOwnerReference(operatorService); err != nil {
 			log.Fatal(err)
 		}
 	}

